src/Disks/ObjectStorages/IObjectStorage.h

```diff
@@ -66,7 +66,32 @@ class IObjectStorage
     virtual bool exists(const StoredObject & object) const = 0;
 
     /// List on prefix, return children (relative paths) with their sizes.
-    virtual void listPrefix(const std::string & path, RelativePathsWithSize & children) const = 0;
+    /// So it should return all "objects" with their sizes.
+    ///
+    /// For example if you do this over filesystem, you should skip folders and
+    /// return files only, so something like on local filesystem:
+    ///
+    ///     find . -type f
+    ///
+    /// NOTE: It makes sense only for real object storages (S3, Azure), since
+    /// it is used only for one of the following:
+    /// - send_metadata (to restore metadata)
+    ///   - see DiskObjectStorage::restoreMetadataIfNeeded()
+    /// - MetadataStorageFromPlainObjectStorage - only for s3_plain disk
+    virtual void listPrefix(const std::string & path, RelativePathsWithSize & children) const;
+
+    /// List on prefix, but it is limited only to current path, something like
+    /// on local filesystem:
+    ///
+    ///     find . -maxdepth 1 -type f
+    ///
+    /// @param path - path to list prefix in
+    /// @param children - list of child nodes
+    /// @param common_prefixes - list of common prefixes in this path, for unix
+    /// filesystem it is simply subfolders.
+    virtual void listPrefixInPath(const std::string & path,
```

Comment:
Can we name those functions in more straightforward way like `findFiles` & `getDirectories()` ?

Reply:
I guess, but I did not want to mix files/folders with S3/GCS/Azure/e.t.c. since they do not have such terms.
And also it is not `findAllFiles` but, `findAllFilesAndDirectories`.
I also thought about adding this is an argument, but came to conclusion that it will be worse.

Reply:
I mean `listPrefix()` can be named `findAllFiles` (it kind of finds all files in a directory recursively) and `listPrefixInPath` can be named `getDirectoryContents()` (it lists both files and subdirectories in a directory).

Reply:
Ok, will rename. I don't have strong feeling about this namings.